util/topgen/intermodule.py

```diff
@@ -343,3 +458,61 @@ def check_intermodule(topcfg: Dict, prefix: str) -> int:
             error += 1
 
     return total_error
+
+
+# Template functions
+def im_defname(obj: OrderedDict) -> str:
+    """return definition struct name
+
+    e.g. flash_ctrl::flash_req_t
+    """
+    if obj["package"] == "":
+        # should be logic
+        return "logic"
+
+    return "{package}::{struct}_t".format(package=obj["package"],
+                                          struct=obj["struct"])
+
+
+def im_topname(obj: OrderedDict, suffix: str = "") -> str:
```

Comment:
do you think this is more / less clear if it's called `netname`? Feels like that's too specific and doesn't cover the `ext` case..but wanted to see what you thought.

Reply:
Agree! Thanks for the suggestion